### PR TITLE
Fix the auto-setup image and test images in CI

### PR DESCRIPTION
**Pull request by tdeebswihart**
## What was changed
I combined the auto-setup and server dockerfiles to work around an issue with (as far as I can tell) `docker buildx bake` and ENTRYPOINTs in multi-stage builds.

I also added a simple test to CI that ensures our images work

## Why?
`docker buildx bake` doesn't seem to let you override the entrypoint of the current build context, so when we changed to using a bakefile we broke the auto-setup image. We can work around this by combining the two into a single multi-stage dockerfile and picking the target we want to build separately.
    
I added two new stages to the server dockerfile to handle this:
    
1. `server` - the server itself without the auto-setup stuff. This is equivalent to what the server.Dockerfile would build prior to this change
2. `auto-setup` - this is the server-with-auto-setup image

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+COMPOSE_PROJECT_NAME=temporal
+CASSANDRA_VERSION=3.11
+ELASTICSEARCH_VERSION=8.0.0
+MYSQL_VERSION=8
+POSTGRESQL_VERSION=13
```
Comment (stephanos) [inline .env:5]: I might have missed it; what step uses this?

**Review by stephanos — APPROVED** [review]

File: .env (hunk 1 above)
Comment (tdeebswihart) [reply .env:5]: These are default values used by docker-compose

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+version: "3.5"
```
Comment (alexshtin) [inline docker-compose.yml:1]: I guess docker-compose repo should be included as submodule here, to make it nice.

File: auto-setup.Dockerfile [hunk 3]
```diff
@@ -1,26 +0,0 @@
-ARG GOPROXY
```
Comment (alexshtin) [inline auto-setup.Dockerfile:1]: Long time ago we had one `Dockerfile`. Then I split it "per image". There was a good reason for it, but I couldn't recall. Why do you merging them back?

Comment (tdeebswihart) [reply auto-setup.Dockerfile:1]: > I combined the auto-setup and server dockerfiles to work around an issue with (as far as I can tell) docker buildx bake and ENTRYPOINTs in multi-stage builds.

Buildx won't let you override the entrypoint the way we needed to in a multi-stage build. I have no idea why, but it doesn't let you do it.

**Pull request merged**